File: Coding/Set3/RotateArray.js
let arr = [1, 3, 5, 7, 9]

rotateArrayRight(arr, 2)
console.log(arr)
rotateArrayLeft(arr, 2)
console.log(arr)

function rotateArrayRight(arr, number_of_times){
    reverseArray(arr, 0, arr.length - 1)
    reverseArray(arr, 0, number_of_times - 1)
    reverseArray(arr, number_of_times, arr.length - 1)
}

function rotateArrayLeft(arr, number_of_times){
    reverseArray(arr, 0, arr.length - 1)
    reverseArray(arr, arr.length - number_of_times, arr.length - 1)
    reverseArray(arr, 0, number_of_times - 1)
}

function reverseArray(arr, startIndex, endIndex){
    while(startIndex < endIndex){
        let temp = arr[startIndex]
        arr[startIndex] = arr[endIndex]
        arr[endIndex] = temp
        startIndex++
        endIndex--
    }
    
}

File: Coding/Java/Set3/RotateArray.java
package Set3;


import java.util.Arrays;

public class RotateArray {

    public static void reverseArray(int[] arr, int start, int end){
        while(start < end){
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }
    public static void rotate_array_right(int[] arr, int number_oftimes){
        reverseArray(arr, 0, arr.length - 1);
        reverseArray(arr, 0, number_oftimes - 1);
        reverseArray(arr, number_oftimes, arr.length - 1);
    }

    public static void rotate_array_left(int[] arr, int number_oftimes){
        reverseArray(arr, 0, arr.length - 1);
        reverseArray(arr, arr.length - number_oftimes, arr.length - 1);
        reverseArray(arr, 0, number_oftimes - 1);
    }

    public static void main(String[] args) {
        int[] arr = {1, 3, 5, 7, 9};
        if(arr.length <= 1){
            System.out.println(Arrays.toString(arr));
            System.exit(0);
        }
        rotate_array_right(arr, 2);
        System.out.println(Arrays.toString(arr));
        rotate_array_left(arr, 2);
        System.out.println(Arrays.toString(arr));
    }
}
